engines/scc_proxy/lib/scc_proxy/engine.rb

```diff
@@ -261,11 +261,16 @@ def announce_system
           auth_header = nil
           auth_header = request.headers['HTTP_AUTHORIZATION'] if request.headers.include?('HTTP_AUTHORIZATION')
           system_information = hwinfo_params[:hwinfo].to_json
-
+          instance_data = hwinfo_params[:hwinfo].fetch(:instance_data, nil) if hwinfo_params[:hwinfo]
```

Comment:
I don't understand why we look into `hwinfo_params[:hwinfo]` for the instance data. 
In my test setup, when running: `suseconnect --url http://172.17.0.1:4224 --instance-data /tmp/idata.xml` it sends this payload: 

```
Parameters: {"hostname"=>"27508a12f2d3", "distro_target"=>"sle-15-x86_64", "instance_data"=>"<instance_data product=\"SUSE\"/>\n", "hwinfo"=>{"arch"=>"x86_64", "container_runtime"=>"docker", "cpus"=>8, "hostname"=>"27508a12f2d3", "mem_total"=>31778, "sockets"=>1, "uname"=>"6.4.0-150600.23.17-default #1 SMP PREEMPT_DYNAMIC Tue Jul 30 06:37:32 UTC 2024 (9c450d7)", "uuid"=>"4c4c4544-0031-3410-8056-b5c04f365a32"}}
```

with instance data in the toplevel, not in `hwinfo`. Does `registercloudguest` do that differently? Do you have a workflow I can test that was failing before and gets fixed by this pr? 

Instead of taking the `'instance_data'` from the `hwinfo` hash in line 69, maybe the better fix would be to not nest in in `hwinfo` in line 338 to avoid confusion.

Reply:
registercloudguest does not change the data, basically registercloudguest generates the instance data per configured command and then writes this data to a file. The file path is passed to SUSEConnect.

https://github.com/SUSE-Enceladus/cloud-regionsrv-client/blob/master/usr/sbin/registercloudguest#L668
https://github.com/SUSE-Enceladus/cloud-regionsrv-client/blob/master/usr/sbin/registercloudguest#L101

This looks like this for module registration:
```/usr/sbin/SUSEConnect --url https://smt-ec2.susecloud.net --product sle-module-desktop-applications/15.6/x86_64 --instance-data /var/cache/cloudregister/1f5090e0-d86f-4132-bfd4-86390b08db4e```

I was trying to get more details about what SUSEConnect does but was not successful even after setting `GODEBUG=http2debug=2` in the environment as instructed when running SUSEConnect with `--debug` 

If someone can spot my typo in the command I'd appreciate it:

```
SUSEConnect --debug --url https://smt-ec2.susecloud.net --instance-data instance_data.txt --product SLES/15.6/x86_64 
```
Theoretically this matches what registercloudguest does, well for the baseproduct registercloudguest does not specify the product. But upon direct invocation the instance data is not transmitted for some reason. The data only contains:

```ser-Agent: SUSEConnect-ng/1.11.0
Content-Length: 351
Accept: application/json,application/vnd.scc.suse.com.v4+json
Accept-Language: POSIX
Authorization: Basic U0NDXzJiMjg5YzBkYjA1ZTQyODdiY2VjN2MxOTFlNTAzMThjOjQ1ODNhNTg
1Mzg1ZjQ4Zjc=
Content-Type: application/json
System-Token: 
Accept-Encoding: gzip

{"hostname":"ip-192-168-10-228","distro_target":"sle-15-x86_64","hwinfo":{"arch"
:"x86_64","cloud_provider":"amazon","cpus":1,"hostname":"ip-192-168-10-228","hyp
ervisor":"xen","mem_total":958,"sockets":1,"uname":"6.4.0-150600.23.17-default #
1 SMP PREEMPT_DYNAMIC Tue Jul 30 06:37:32 UTC 2024 (9c450d7)","uuid":"ec258f7a-9
c5e-f1a7-27cc-34dfe1af340f\n"}}
```

Reply:
Anyway, taking a look at SUSEConnect [1] indicates that the instance data is not part of hwinfo.

[1] https://github.com/SUSE/connect-ng/blob/main/internal/connect/api.go#L241

Reply:
The request where this happens is to activate the product, that request has params that do not include `instance_data`:
```bash
\"identifier\"=>\"sles-LTSS\", \"version\"=>\"15.4\", \"arch\"=>\"x86_64\", \"release_type\"=>\"\", \"token\"=>\"some_reg_code\", \"email\"=>\"\", \"format\"=>\"json\", \"controller\"=>\"api/connect/v4/systems/products\", \"action\"=>\"activate\", \"product\"=>{\"release_type\"=>\"\", \"identifier\"=>\"sles-LTSS\", \"version\"=>\"15.4\", \"arch\"=>\"x86_64\"}}
```

That said, better fix is to adapt the created params to announce the system in a different to include `instance_data` outside `hwinfo_params` from [this](https://github.com/SUSE/rmt/blob/master/engines/scc_proxy/lib/scc_proxy/engine.rb#L332)
to 
```ruby
params['instance_data'] = @system.instance_data
```
and keep the `get_instance_data` as it was originally
for `BYOS` it is not that important, as instance verification will fail anyway regardless of where the data is
but the structure would be correct, IMO